gnat2goto/ireps/ireps_generator.py

```diff
@@ -1609,46 +1636,28 @@ def generate_code(self, optimize, schema_file_names):
         write(b, "is")
         continuation(b)
         with indent(b):
-            write(b, "S : constant String := Integer'Image (I);")
+            write(b, "Int_As_Str : constant String := Integer'Image (I);")
+            write(b, "S : constant String := (if I >= 0 then")
+            with indent(b):
+                write(b, "Int_As_Str (2 .. Int_As_Str'Last)")
+            write(b, "else")
+            with indent(b):
+                write(b, "Int_As_Str);")
         write(b, "begin")
         with indent(b):
-            write(b, "return V : constant JSON_Value := Create_Object do")
-            with indent(b):
-                write(b, "if I >= 0 then")
-                with indent(b):
-                    write(b, 'V.Set_Field ("id", S (2 .. S\'Last));')
-                write(b, "else")
-                with indent(b):
-                    write(b, 'V.Set_Field ("id", S);')
-                write(b, "end if;")
-            write(b, "end return;")
+            write(b, "return Trivial_Irep (S);")
         write(b, "end Trivial_Integer;")
         write(b, "")
 
         write_comment_block(b, "Trivial_String")
         write(b, "function Trivial_String (S : String_Id) return JSON_Value")
-        write(b, "is")
-        continuation(b)
-        write(b, "begin")
         with indent(b):
-            write(b, "String_To_Name_Buffer (S);")
-            write(b, "return V : constant JSON_Value := Create_Object do")
-            with indent(b):
-                write(b, 'V.Set_Field ("id",       Name_Buffer (1 .. Name_Len));')
-            write(b, "end return;")
-        write(b, "end Trivial_String;")
+            write(b, "renames Trivial_Irep;")
```

Comment:
Renamed for now, but I'll probably just go over and remove remaining references to `Trivial_String` in the future.